src/main/java/edu/usf/cutr/gtfs_realtime/bullrunner/GtfsRealtimeProviderImpl.java

```diff
@@ -202,11 +203,74 @@ public void stop() {
+        Iterator IT = _providerConfig.ExternalIDMap.entrySet().iterator();
+        while (IT.hasNext()){
+            Map.Entry pair = (Map.Entry) IT.next();
+            String route_id = pair.getKey().toString();
+            String external_id = pair.getValue().toString();
+
+            // get vehicle locations for this route_id:
+            JSONArray vehicleArray = downloadVehicleDetails(external_id);
+
+            // loop through vehicleArray to build vehiclePosition for the given route
+            for (int k = 0; k < vehicleArray.length(); k++){
+
+
+                JSONObject vehicleObj = vehicleArray.getJSONObject(k);
+                // initiate feed
+                TripDescriptor.Builder tripDescriptor = TripDescriptor.newBuilder();
+                Position.Builder position = Position.newBuilder();
+                VehicleDescriptor.Builder VehicleInfo = VehicleDescriptor.newBuilder();
+                FeedEntity.Builder vehiclePositionEntity = FeedEntity.newBuilder();
+
+                // set values for feed
+                tripDescriptor.setRouteId(route_id);
+                position.setBearing((float) vehicleObj.getDouble("headingDegrees"));
+                position.setLatitude((float) vehicleObj.getDouble("lat"));
+                position.setLongitude((float) vehicleObj.getDouble("lon"));
+                position.setSpeed((float) vehicleObj.getDouble("speed"));
+                VehicleInfo.setId(String.valueOf(vehicleObj.getInt("id")));
+                VehicleInfo.setLabel(vehicleObj.getString("name"));
+
+                // Build feed
+                VehiclePosition.Builder VehiclePosition_route = VehiclePosition.newBuilder();
+                VehiclePosition_route.setPosition(position);
+                VehiclePosition_route.setTrip(tripDescriptor);
+                VehiclePosition_route.setVehicle(VehicleInfo);
+                VehiclePosition_route.setTimestamp(
+						Instant.parse(vehicleObj.getString("lastUpdated")).getEpochSecond()
+				);
+                if (vehicleObj.getDouble("passengerLoad") <= 0) VehiclePosition_route.setOccupancyStatus( OccupancyStatus.EMPTY );
+                else if (vehicleObj.getDouble("passengerLoad") <= 50) VehiclePosition_route.setOccupancyStatus( OccupancyStatus.MANY_SEATS_AVAILABLE );
+                else if (vehicleObj.getDouble("passengerLoad") <= 70) VehiclePosition_route.setOccupancyStatus( OccupancyStatus.FEW_SEATS_AVAILABLE );
+                else if (vehicleObj.getDouble("passengerLoad") <= 90) VehiclePosition_route.setOccupancyStatus( OccupancyStatus.STANDING_ROOM_ONLY );
+                else if (vehicleObj.getDouble("passengerLoad") <= 95) VehiclePosition_route.setOccupancyStatus( OccupancyStatus.CRUSHED_STANDING_ROOM_ONLY );
+                else VehiclePosition_route.setOccupancyStatus( OccupancyStatus.FULL );
+
+                vehicleFeedID ++ ;
+                vehiclePositionEntity.setId(Integer.toString(vehicleFeedID));

```

Comment:
Is there an ID field from the proprietary API that we could use here instead of a simple counter value?  The vehicle ID should persist between updates, and there is no guarantee here from what I can see that this will happen.

Reply:
This is the Feed Entity ID. Vehicle ID is set in [line 238](https://github.com/minhhpham/bullrunner-gtfs-realtime-generator/commit/5abb1fbef754f1d61b12bdce464c47d7430cf9bf#diff-dab25dc89fc5315c313717a2e848d053R238).